File: bonapetit.js
function bonAppetit(bill, k, b) {
  var totalSharedCost = 0;
  var personAfterSharedAmount = 0;

  for (var i = 0; i < bill.length; i++) {
    if (k == i) {
      continue;
    } else {
      totalSharedCost += bill[i];
    }
  }

  var personAfterSharedAmount = totalSharedCost / 2;

  var banShouldPay = b - personAfterSharedAmount;

  if (banShouldPay) {
    return "Bon Appetit";
  } else {
    return banShouldPay;
  }
}

console.log(bonAppetit(4, [3, 10, 2, 9], 12));

File: bonapetit.rb
# frozen_string_literal: true

def bonAppetit(bill, k, b)
  totalSharedCost = 0

  (0...bill.size).each do |i|
    totalSharedCost += bill[i] if k != i
  end

  banShouldPay = b - (totalSharedCost / 2)

  if banShouldPay == 0
    puts 'Bon Appetit'
  else
    p banShouldPay
  end
end

p bonAppetit([3, 10, 2, 9], 1, 12)
p bonAppetit([3, 10, 2, 9], 1, 7)
